### ISSUE TITLE
Add (back) support sorting by sortText and/or label

### ISSUE THREAD

**Issue by lithammer** [nvim-lua/completion-nvim#234]
It's currently possible to sort by 'alphabet', 'length' or 'none'. Neither are actually very useful (to me), and instead we should add back the sorting as provided by Neovim and ideally make it the default (instead of 'alphabet'):

```lua
--@private
--- Sorts by CompletionItem.sortText.
---
--@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
local function sort_completion_items(items)
  table.sort(items, function(a, b)
    return (a.sortText or a.label) < (b.sortText or b.label)
  end)
end
```
https://github.com/neovim/neovim/blob/a260d5def363ca30f0473365ea514f2dedd0d0ba/runtime/lua/vim/lsp/util.lua#L297-L305

This more closely resembles how VS Code is sorting completion items. And seems to be the recommended way according to the [spec](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion).

**Comment by lithammer** [comment]
I would be willing to help out with a pull request as well, if you'd like.

**Comment by lithammer** [comment]
Something like this together with `let g:completion_sorting = 'none'` is probably the ideal default to me:

```diff
diff --git a/lua/completion/source/lsp.lua b/lua/completion/source/lsp.lua
index 87a4641..6352649 100644
--- a/lua/completion/source/lsp.lua
+++ b/lua/completion/source/lsp.lua
@@ -66,6 +66,12 @@ local function get_context_aware_snippets(item, completion_item, line_to_cursor)
   end
 end
 
+local function sort_completion_items(items)
+  table.sort(items, function(a, b)
+    return (a.sortText or a.label) < (b.sortText or b.label)
+  end)
+end
+
 local function text_document_completion_list_to_complete_items(result, params)
   local items = vim.lsp.util.extract_completion_items(result)
   if vim.tbl_isempty(items) then
@@ -74,7 +80,7 @@ local function text_document_completion_list_to_complete_items(result, params)
 
   local customize_label = opt.get_option('customize_lsp_label')
   -- items = remove_unmatch_completion_items(items, prefix)
-  -- sort_completion_items(items)
+  sort_completion_items(items)
 
   local matches = {}
 
```

**Comment by lithammer** [comment]
Fixed by #330
